module-check/src/main/clojure/clojure/core/typed/base_env.clj

```diff
@@ -1443,7 +1443,17 @@ clojure.core/rand-nth (All [x] [(U (Indexed x) (SequentialSeqable x)) -> x])
   this-ns
 clojure.set/union (All [x] [(Set x) * -> (Set x)])
 clojure.set/intersection (All [x] [(Set x) (Set x) * -> (Set x)])
+
 clojure.set/difference (All [x] [(Set x) (Set Any) * -> (Set x)])
+clojure.set/subset? [(Set Any) (Set Any) -> Boolean]
+clojure.set/superset? [(Set Any) (Set Any) -> Boolean]
+clojure.set/join [(Set (Map Any Any)) (Set (Map Any Any)) -> (Set (Map Any Any))]
+clojure.set/index [(Set (Map Any Any)) (Vec Any) -> (Map (Map Any Any) (Set (Map Any Any)))]
```

Comment:
I think you're referring to the comment for line 1450 (about `join`), right? Well, seems like join creates new maps by merging maps from first and second arguments so some sense all maps in return values are different from arguments.